### PR TITLE
Introduce LITMUS-RT guest into prod-devel

### PR DESCRIPTION
**Pull request by aanisov**

### CODE REVIEW

File: recipes-domu/domu-image-litmusrt/domu-image-litmusrt.bbappend [hunk 1]
```diff
@@ -0,0 +1,53 @@
+
```
Comment (andr2000) [inline recipes-domu/domu-image-litmusrt/domu-image-litmusrt.bbappend]: no need for an empty line here?

File: recipes-domu/domu-image-litmusrt/domu-image-litmusrt.bbappend [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+# these layers will be added to bblayers.conf on do_configure
+XT_QUIRK_BB_ADD_LAYER += " \
+    meta-litmusrt \
+"
+
+# Set default poky revision to pyro
+BRANCH ?= "pyro"
```
Comment (andr2000) [inline recipes-domu/domu-image-litmusrt/domu-image-litmusrt.bbappend]: do you expect yet another bbappend which might overwrite this?
?= -> =

File: recipes-domu/domu-image-litmusrt/domu-image-litmusrt.bbappend [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+python do_configure_append() {
+    bb.build.exec_func("configure_versions_base", d)
+}
+
+################################################################################
+# Generic
+################################################################################
+XT_BB_IMAGE_TARGET = "litmusrt-image-minimal"
+
+
```
Comment (andr2000) [inline recipes-domu/domu-image-litmusrt/domu-image-litmusrt.bbappend]: extra line

File: recipes-domu/domu-image-litmusrt/files/meta-xt-prod-extra/recipes-extended/images/litmusrt-image-minimal.bbappend [hunk 4]
```diff
@@ -0,0 +1,10 @@
+require inc/xt_shared_env.inc
+
+
```
Comment (andr2000) [inline recipes-domu/domu-image-litmusrt/files/meta-xt-prod-extra/recipes-extended/images/litmusrt-image-minimal.bbappend]: extra line

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/domr-install-artifacts/domr-install-artifacts.bb [hunk 5]
```diff
@@ -0,0 +1,20 @@
+LICENSE = "CLOSED"
+
+require inc/xt_shared_env.inc
+
+FILES_${PN} += "${base_prefix}${XT_DIR_ABS_ROOTFS_DOMR}"
+
+# FIXME: if not forced and sstate cache is used then an old version of
+# this package (read old DomU kernel images) can be used from cache
```
Comment (andr2000) [inline recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/domr-install-artifacts/domr-install-artifacts.bb]: pls either remove DomU or put DomR

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/domr-install-artifacts/domr-install-artifacts.bb [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+# regardless of the fact that binaries may have actually changed, e.g.
+# the recipe code is not changed, there is no SRC_URI with checksum
+# force install so if DomU kernel changes we use the latest binaries
```
Comment (andr2000) [inline recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/domr-install-artifacts/domr-install-artifacts.bb]: ditto

File: recipes-domu/domu-image-litmusrt/files/meta-xt-prod-extra/recipes-extended/images/litmusrt-image-minimal.bbappend [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+share_artifacts () {
+    mkdir -p ${XT_DIR_ABS_SHARED_BOOT_DOMR}
+    rm -f ${XT_DIR_ABS_SHARED_BOOT_DOMR}/*.cpio.gz
+    cp -f --no-dereference --preserve=links ${IMGDEPLOYDIR}/*.cpio.gz ${XT_DIR_ABS_SHARED_BOOT_DOMR}
+}
+
+IMAGE_POSTPROCESS_COMMAND += " share_artifacts ; "
```
Comment (andr2000) [inline recipes-domu/domu-image-litmusrt/files/meta-xt-prod-extra/recipes-extended/images/litmusrt-image-minimal.bbappend]: do you need space before ";"?

**Comment by aanisov** [comment]
updated

**Comment by aanisov** [comment]
@iartemenko be so kind to merge this

**Pull request merged**